localstack/services/s3/provider.py

```diff
@@ -255,6 +255,16 @@ def get_bucket_location(
         self, context: RequestContext, bucket: BucketName, expected_bucket_owner: AccountId = None
     ) -> GetBucketLocationOutput:
         response = call_moto(context)
+        # LocationConstraint is a bit of weird response, we return directly XML from moto because it does not have
+        # a GetBucketLocationOutput XML tag.
+        # some tools (Serverless) require a newline after the "<?xml ...>\n" preamble line, e.g., for LocationConstraint
+        # this is required because upstream moto is generally collapsing all S3 XML responses
```

Comment:
This comment is a bit confusing to be honest.
The key essence is:
- When implementing the ASF provider, this operation is implemented because:
  - The spec defines a root element `GetBucketLocationOutput` containing a `LocationConstraint` member, where S3 actually just returns the `LocationConstraint` on the root level.
  - We circumvent the root level element here by patching the spec such that this operation returns a single "payload", which causes the serializer to directly take the payload element.
- The above "hack" causes the fix in the serializer to not be picked up here, which is why we need to manually do this here.

Reply:
I'll update 👍 thanks for the synthesis of the issue, it's much better than what I could come up with 😅